src/lib.rs

```diff
@@ -1,3 +1,114 @@
+//! # Floem
+//! Floem is cross-platform GUI framework for Rust 🦀. It aims to be extremely performant while providing world-class developer ergonomics.
+//!
+//! ## Views
+//! Floem models the UI using a tree of [Views](view::View) that are built once. `Views` react to state changes and events.
+//! Views ar self-contained components that can be composed together to create complex UIs.
+//!
+//! ## Ids and Id paths
+//! [Id](id::Id)s are unique identifiers for views. They're used to identify views in the view tree. These ids are assigned via the [AppContext](context::AppContext) and are unique across the entire application.
+//!
+//! ## Events
+//! Events are passed down from the window to the view tree. Each view can handle events and propagate them to its children.
+//! Events can be passed to a specific view using its [Id](id::Id) path. [Id](id::Id) paths are unique ways to quickly identify a view, its children and its parent in the tree.
+//!
+//! Floem is responsible for making a view active. for handling drag and drop mechanics
+//! It's a View's responsibility to handle events and decide whether they're handled.
+//!
+//! ## State management  
+//!
+//! You will probably want your view components to have some state. You should place any state that affects
+//! you view inside a signal so that you can react to updates and update the `View`. Signals are reactive values that can be read from and written to.
+//!
+//! ### Use state to update your view
+//!
+//! To affect the layout and rendering of your component, you will need to send a state update to your component with [Id::update_state](id::Id::update_state)
```

Comment:
I think in general most time (especially for anything not super complex) developing floem will be done by combining pre-existing views provided by floem and not creating completely new ones. I think the top level documentation should reflect that and focus on composing views. I think that is a more user friendly introduction as well.

Reply:
Cool, I will rearrange to reflect a higher-level approach. I think this "how to build views" stuff could fit nicely at the top of `view.rs` with an "Advanced" header/section here that links to that.